### PR TITLE
Stateless NAT and notrack support

### PR DESCRIPTION
**Pull request by alexispires**
This PR adds support for ``notrack`` expression

### CODE REVIEW

**Comment by alexispires** [comment]
Relate to #80

File: expr/notrack.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"github.com/mdlayher/netlink"
+	"golang.org/x/sys/unix"
+)
+
+type Notrack struct{}
+
+func (e *Notrack) marshal() ([]byte, error) {
+
```
Comment (stapelberg) [inline expr/notrack.go]: Can you remove the blank line please?

**Comment by alexispires** [comment]
@stapelberg i just added stateless nat in this pr, can you review it?

File: expr/payload.go [hunk 2]
```diff
@@ -31,20 +32,57 @@ const (
...
-	data, err := netlink.MarshalAttributes([]netlink.Attribute{
-		{Type: unix.NFTA_PAYLOAD_DREG, Data: binaryutil.BigEndian.PutUint32(e.DestRegister)},
-		{Type: unix.NFTA_PAYLOAD_BASE, Data: binaryutil.BigEndian.PutUint32(uint32(e.Base))},
-		{Type: unix.NFTA_PAYLOAD_OFFSET, Data: binaryutil.BigEndian.PutUint32(e.Offset)},
-		{Type: unix.NFTA_PAYLOAD_LEN, Data: binaryutil.BigEndian.PutUint32(e.Len)},
-	})
+
+	var attrs []netlink.Attribute
+
+	if e.SourceRegister > 0 {
```
Comment (stapelberg) [inline expr/payload.go]: Don’t we need to distinguish between an unset SourceRegister field and a 0 SourceRegister here, like in the previous PR?

Comment (alexispires) [reply expr/payload.go]: sure

File: expr/payload.go [hunk 3]
```diff
@@ -31,20 +33,64 @@ const (
...
 )
 
+// Possible PayloadCsumType values.
+const (
+	CsumTypeNone PayloadCsumType = unix.NFT_PAYLOAD_CSUM_NONE
+	CsumTypeInet PayloadCsumType = unix.NFT_PAYLOAD_CSUM_INET
+)
+
+// Possible PayloadType values.
+const (
```
Comment (alexispires) [inline expr/payload.go]: Related to (in netlink message):
[ payload write reg 1 => 4b @ network header + 16 ]
[ payload load 4b @ network header + 12 => reg 2 ]

File: expr/payload.go [hunk 4]
```diff
@@ -31,20 +33,64 @@ const (
...
+	PayloadLoad PayloadOperationType = iota
+	PayloadWrite
```
Comment (alexispires) [inline expr/payload.go]: Related to (in netlink message):
[ payload write reg 1 => 4b @ network header + 16 ]
[ payload load 4b @ network header + 12 => reg 2 ]

**Pull request merged**